test/support/builders/atom_watcher_event.js

```diff
@@ -0,0 +1,70 @@
+const metadata = require('../../../core/metadata')
+const events = require('../../../core/local/steps/event')
+
+/*::
+import type { Stats } from 'fs'
+import type { AtomWatcherEvent, Action, Kind, Batch } from '../../../core/local/steps/event'
+*/
+
+function randomPick (elements /*: Array<T> */) /*: T */{
+  const l = elements.length
+  const i = Math.floor(Math.random() * Math.floor(l))
+  return elements[i]
+}
+
+module.exports = class AtomWatcherEventBuilder {
+  constructor () {
+    this._event /*: AtomWatcherEvent */ = {
+      action: randomPick(events.ACTIONS),
+      kind: randomPick(events.KINDS),
+      path: '/',
+      _id: '/'
+    }
+  }
+
+  build () /*: AtomWatcherEvent */ {
+    return this._event
+  }
+
+  action (newAction /*: Action */) /*: this */ {
+    if (!events.ACTIONS.includes(newAction)) {
+      throw new Error(`Unknown event action "${newAction}". Available actions: ${events.ACTIONS.join(', ')}.`)
+    }
+    this._event.action = newAction
+    return this
+  }
+
+  kind (newKind /*: Kind */) /*: this */ {
+    if (!events.KINDS.includes(newKind)) {
+      throw new Error(`Unknown event kind "${newKind}". Available kinds: ${events.KINDS.join(', ')}.`)
+    }
+    this._event.kind = newKind
+    return this
+  }
+
+  path (newPath /*: string */) /*: this */ {
+    this._event.path = newPath
+    this._event._id = metadata.id(newPath)
```

Comment:
This is generated by `add_infos`, what do we want as a goal?

Reply:
I think this builder may be used for testing most of the steps, but not for `addInfos`

Reply:
Maybe I should make some changes to the builder to make it useful when testing `addInfos`?!

Reply:
@taratatach Did you use all these methods in your tests? Or are we trying to anticipate future needs?

Reply:
You can see some examples in https://github.com/cozy-labs/cozy-desktop/pull/1377